waspc/src/Wasp/Util.hs

```diff
@@ -67,12 +68,22 @@ headSafe :: [a] -> Maybe a
 headSafe [] = Nothing
 headSafe xs = Just (head xs)
 
+second :: (b -> d) -> (a, b, c) -> (a, d, c)
+second f (x, y, z) = (x, f y, z)
+
 jsonSet :: Text.Text -> Aeson.Value -> Aeson.Value -> Aeson.Value
 jsonSet key value (Aeson.Object o) = Aeson.Object $ M.insert key value o
 jsonSet _ _ _ = error "Input JSON must be an object"
 
 indent :: Int -> String -> String
-indent numSpaces = intercalate "\n" . map (replicate numSpaces ' ' ++) . splitOn "\n"
+indent numSpaces = intercalate "\n" . map (stripIfEmpty . (indentation ++)) . splitOn "\n"
+  where
+    indentation = replicate numSpaces ' '
+
+stripIfEmpty :: String -> String
+stripIfEmpty str
+  | null $ Text.unpack $ Text.strip $ Text.pack str = ""
```

Comment:
I would skip all the Text packing and unpacking, and instead do smth like
```hs
import Data.Char (isSpace)

stripIfEmpty :: String -> String
stripIfEmpty str | all isSpace str = ""
                 | otherwise = str
```

Reply:
Also, would rename it to something else, as concept of stripping if empty is weird. Also, what does empty mean? I would intuitively expect empty to be "", not string consisting of white spaces.

Maybe `toEmptyStringIfAllWhiteSpace`?